appservice/src/SiteWrapper.ts

```diff
@@ -293,6 +298,43 @@ export class SiteWrapper {
         await requestP.get(`https://${this.defaultHostName}/admin/host/status`, requestOptions);
     }
 
+    public async validateDeployment(outputChannel: vscode.OutputChannel, pollingInterval: number = 5000, pollingAttempts: number = 12): Promise<responseStatus[]> {
+        // by default, try for a minute before we declare that the deployment failed
+        // tslint:disable-next-line:no-http-string
+        const uri: string = `${this.isSsl ? 'https://' : 'http://'}${this.defaultHostName}`;
+        const validatingDeployment: string = `Validating your ${this.isFunctionApp ? 'function' : 'web'} app successfully deployed.`;
+        const options: {} = {
+            method: 'GET',
+            uri: uri,
+            resolveWithFullResponse: true
+        };
+        let response: IncomingMessage;
+        const statuses: responseStatus[] = [];
+        for (let i: number = 0; i < pollingAttempts; i += 1) {
+            try {
+                await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
```

Comment:
I'd move the pollingInterval code to the bottom of the loop. You don't want to force the user to wait 5 seconds if the site is ready immediately (although I realize its unlikely to be ready immediately 😂)